### PR TITLE
Xhr

### PR DESCRIPTION
**Pull request by guvenkaranfil**

### CODE REVIEW

File: src/net/xhr/xhr.android.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      return '';
+    }
+    return this._response ? this._response.toString() : '';
+  }
+
+  public get responseType(): ResponseTypes {
+    return this._responseType;
+  }
+
+  public set responseType(value: ResponseTypes) {
```
Comment (muhammedyalcin) [inline src/net/xhr/xhr.android.ts:116]: please control other unsupported types

**Review by muhammedyalcin — CHANGES_REQUESTED** [review]

File: src/net/xhr/common.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+export const XMLHttpRequestResponseType = {
```
Comment (furkanarabaci) [inline src/net/xhr/common.ts:1]: change the name of this file to `xhr.ts`. common.ts will be confusing per architecture we have.

Comment (furkanarabaci) [reply src/net/xhr/common.ts:1]: If those values will be used everywhere, move it to `shared` folder.

If they are gonna use globally on the project as type, we should move those to `types/` folder

File: src/net/xhr/index.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    readonly responseText: string;
+    responseType: ResponseTypes;
+    readonly responseURL?: string;
+    readonly status: number;
+    readonly statusText: string;
+    timeout: number;
+    readonly upload : XMLHttpRequestEventTarget;
+}
+
+const XHR: ConstructorOf<IXHR, Partial<IXHR>> = require(`./xhr.${Device.deviceOS.toLowerCase()}`).default;
```
Comment (furkanarabaci) [inline src/net/xhr/index.ts:47]: Move everything except this line and export defaul to `xhr.ts` to not risk circular require on projects

File: src/net/xhr/xhr.android.ts [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+  extends NativeEventEmitterComponent<TEvent | XHREventsEvents, any, TProps>
+  implements IXHR
+{
+  public static UNSENT = 0;
+  public static OPENED = 1;
+  public static HEADERS_RECEIVED = 2;
+  public static LOADING = 3;
+  public static DONE = 4;
+
+  public onabort: (...args: any[]) => void = () => {};
```
Comment (furkanarabaci) [inline src/net/xhr/xhr.android.ts:21]: `public` key is redundant on Typescript, you can safely remove every public keyword

**Comment by furkanarabaci** [comment]
emitEvent stuff should be changed, we can prioritize that on EventEmitter side.

**Review by furkanarabaci — CHANGES_REQUESTED** [review]

File: src/net/xhr/xhr.ios.ts [hunk 5]
```diff
@@ -0,0 +1,384 @@
+import { HTTPRequestMethods, IXHR } from '.';
+
+import NativeEventEmitterComponent from '../../core/native-event-emitter-component';
+import { MobileOSProps } from '../../core/native-mobile-component';
+import { ResponseTypes, statuses, XMLHttpRequestResponseType } from './common';
+import { FormData, FormDataPart } from './common';
+import { XHREventsEvents } from './xhr-events';
+
+class XHR<TEvent extends string = XHREventsEvents, TProps extends MobileOSProps = MobileOSProps> extends NativeEventEmitterComponent<TEvent | XHREventsEvents, any, TProps> implements IXHR {
```
Comment (furkanarabaci) [inline src/net/xhr/xhr.ios.ts:9]: Change the class name to XHRIOS

File: src/net/xhr/xhr.ios.ts [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+  private _errorFlag: boolean;
+  private _sendFlag: boolean;
+  private _responseType: ResponseTypes = 'text';
+  private _responseURL?: string;
+  private _status: number;
+
+  private _listeners: Map<string, Array<Function>> = new Map<string, Array<Function>>();
+
+  protected createNativeObject() {
+    this._readyState = XHR.UNSENT;
```
Comment (furkanarabaci) [inline src/net/xhr/xhr.ios.ts:39]: Move this statement to init() or constructor

**Review by furkanarabaci — CHANGES_REQUESTED** [review]

**Pull request merged**